File: src/native/printer-plugin/registry/driver-registry.test.ts
import { describe, it, expect } from 'vitest'
import { DriverRegistry } from './driver-registry'
import type { PrinterDriver } from '../internal/types'

class FakeDriver implements PrinterDriver {
  buildTestPrintBytes(): Uint8Array {
    return new Uint8Array([0x01])
  }
}

describe('DriverRegistry', () => {
  it('resolves a registered driver by type', () => {
    const registry = new DriverRegistry()
    registry.register('generic-escpos', () => new FakeDriver())

    const driver = registry.resolve('generic-escpos')

    expect(driver).toBeInstanceOf(FakeDriver)
  })

  it('throws a clear error when the type is not registered', () => {
    const registry = new DriverRegistry()
    expect(() => registry.resolve('epson')).toThrow('Driver "epson" chưa được đăng ký.')
  })
})

File: android/app/src/test/java/com/ndtcore/pos/printer/registry/DriverRegistryTest.kt
package com.ndtcore.pos.printer.registry

import com.ndtcore.pos.printer.driver.PrinterDriver
import org.junit.Assert.assertTrue
import org.junit.Assert.assertThrows
import org.junit.Test

private class FakeDriver : PrinterDriver {
    override fun buildTestPrintBytes(): ByteArray = byteArrayOf(0x01)
}

class DriverRegistryTest {
    @Test
    fun `resolve returns a registered driver`() {
        val registry = DriverRegistry()
        registry.register("generic-escpos") { FakeDriver() }

        assertTrue(registry.resolve("generic-escpos") is FakeDriver)
    }

    @Test
    fun `resolve throws a clear error when the type is not registered`() {
        val registry = DriverRegistry()
        val error = assertThrows(IllegalArgumentException::class.java) { registry.resolve("epson") }
        assertTrue(error.message!!.contains("epson"))
    }
}
